dot/peerset/handler.go

```diff
@@ -5,12 +5,15 @@ package peerset
 
 import (
 	"context"
+	"sync"
 
 	"github.com/libp2p/go-libp2p-core/peer"
 )
 
 // Handler manages peerSet.
 type Handler struct {
+	mu sync.Mutex
```

Comment:
why do we need a mutex for the channel?

Reply:
Yes I'm curious as well. I get why for the `ps.nodes` map but not for the channel.

Reply:
The race condition tests were accusing a goroutine writing in the channel while another goroutine was closing the channel

Reply:
Hmmm.. that sounds like we need to refactor the code to avoid this.  A single producer (writer & closer) to the channel will alleviate this issue.